forge-game/src/main/java/forge/game/ability/effects/PlaneswalkEffect.java

```diff
@@ -18,13 +19,20 @@ public void resolve(SpellAbility sa) {
         }
 
         if (!sa.hasParam("DontPlaneswalkAway")) {
-            for (Player p : game.getPlayers()) {
-                p.leaveCurrentPlane();
+            for (Player p : game.getPlayers()) {                
+            if (sa.hasParam("Optional") && !p.getController().confirmAction(sa, null,
+                Localizer.getInstance().getMessage("lblWouldYouLikeToPlaneswalk"), null)) {
+                    return;
+            }
+            else {              
+            	p.leaveCurrentPlane();
+            }
             }
```

Comment:
This formatting looks wonky

Line 23 should happen before we run the player loop... instead of using `p.getController()` it should probably be `sa.getActivatingPlayer().getController()`